### PR TITLE
Enable nullable in TestHost and Hosting.WindowsServices

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspnetcore/issues/5680

### CODE REVIEW

File: src/Hosting/TestHost/src/HttpContextBuilder - Copy.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+
+#nullable enable
+
+namespace Microsoft.Extensions.Hosting
+{
+    internal class HostFactoryResolver
```
Comment (JamesNK) [inline src/Hosting/TestHost/src/HttpContextBuilder - Copy.cs:12]: HostFactoryResolver.cs is shared source from a NuGet package. PR to update NuGet package here - https://github.com/dotnet/runtime/pull/45513

Local copy is temporary.

File: src/Hosting/WindowsServices/src/WebHostService.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public class WebHostService : ServiceBase
         /// Creates an instance of <c>WebHostService</c> which hosts the specified web application.
         /// </summary>
         /// <param name="host">The configured web host containing the web application to host in the Windows service.</param>
-        public WebHostService(IWebHost host)
+        public WebHostService(IWebHost? host)
```
Comment (Tratcher) [inline src/Hosting/WindowsServices/src/WebHostService.cs:25]: ? This API throws if the host is null.

File: src/Hosting/WindowsServices/src/WebHostWindowsServiceExtensions.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public static class WebHostWindowsServiceExtensions
         ///         }
         ///     </code>
         /// </example>
-        public static void RunAsService(this IWebHost host)
+        public static void RunAsService(this IWebHost? host)
```
Comment (Tratcher) [inline src/Hosting/WindowsServices/src/WebHostWindowsServiceExtensions.cs:36]: Also should not be null.

**Review by Tratcher — APPROVED** [review]

**Comment by JamesNK** [comment]
Blocked on https://github.com/dotnet/runtime/pull/45513

**Pull request merged**